src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -64,10 +64,45 @@ public override BoundNode Visit(BoundNode node)
                 if (operation != null)
                 {
                     this.nodes.Add(operation);
+                    switch (operation.Kind)
+                    {
+                        case OperationKind.InvocationExpression:
+                            nodes.AddRange(((IInvocationExpression)operation).ArgumentsInSourceOrder);
+                            break;
+                        case OperationKind.ObjectCreationExpression:
+                            var objCreationExp = (IObjectCreationExpression)operation;
+                            nodes.AddRange(objCreationExp.ConstructorArguments);
+                            nodes.AddRange(objCreationExp.MemberInitializers);
+                            break;
+                    }
                 }
-
                 return base.Visit(node);
             }
+
+            // We skip visiting all `BoundLocalDeclaration` nodes in `BoundMultipleLocalDeclarations` 
+            // (but not their children), since they are not statements in this case.
+            public override BoundNode VisitMultipleLocalDeclarations(BoundMultipleLocalDeclarations node)
+            {
+                foreach (var decl in node.LocalDeclarations)
+                {
+                    this.Visit(decl.DeclaredType);
+                    this.Visit(decl.InitializerOpt);
+                    this.VisitList(decl.ArgumentsOpt);
+                }
+                return null;
+            }
+
+            // Skip visiting `BoundAssignmentOperator` nodes (but not their children) if they are used for initializing members in object creation.
+            // The corresponding operations are covered by `IMemberInitializer`.
+            public override BoundNode VisitObjectInitializerExpression(BoundObjectInitializerExpression node)
+            {
+                foreach (BoundAssignmentOperator assignment in node.Initializers)
```

Comment:
It looks like the cast to BoundAssignmentOperator is not safe. Some Initializers might not be BoundAssignmentOperator nodes in an error scenario.